src/pages/RegPage.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useReducer, useState } from "react";
+import DisplayRequest from "../views/DisplayRequest";
+import FlowData from "../views/FlowData";
+import API, { getFormattedURL, getFormattedProxyURL } from "../API";
+import RegistrationProvider from "../util/RegistrationProvider";
+import { Container, Row, Col } from "react-bootstrap";
+import Registration from "../views/Registration";
+
+const RegPage = (props) => {
+    const reset = () => {
+        setRequest({});
+        setResponse({});
+        setCSRF("");
+        setFlowID("");
+        setSessionID("");
+    }
+
+    const resetPrint = () => {
+        setRequest({});
+        setResponse({});
+    }
+
+    const [request, setRequest] = useState({});
+
+    const [response, setResponse] = useState({});
+
+    const [flowID, setFlowID] = useState("");
+    const [csrf, setCSRF] = useState("");
+    const [accountKey, setAccountKey] = useState("");
+    const [sessionID, setSessionID] = useState("");
+
+    const initRegistration = async () => {
+        reset();
+
+        let registration = new RegistrationProvider(props.cluster, props.flow);
+        let newRequest = registration.getInitRequest();
+
+        setRequest(newRequest);
+
+        var jsonRes = await API.makeRequest(newRequest);
+        setResponse({ ...response, headers: jsonRes.headers, status: jsonRes.status, body: jsonRes.json });
+        setFlowID(jsonRes.json.id);
+        setCSRF(jsonRes.json.ui.nodes.filter(x => x.attributes.name == "csrf_token").map(x => x.attributes.value)[0]);
+    }
+
+    const submitRegistration = async () => {
+        resetPrint();
+
+        let registration = new RegistrationProvider(props.cluster, props.flow);
+        let newRequest = registration.getSubmitRequest(flowID, csrf);
+
+        setRequest(newRequest);
+
+        console.log(newRequest);
+
+        var jsonRes = await API.makeRequest(registration.stringifyBody(newRequest));
+        setResponse({ ...response, headers: jsonRes.headers, status: jsonRes.status, body: jsonRes.json });
+        setAccountKey(jsonRes.json.identity.traits.accountKey);
+        setSessionID(jsonRes.json.session.id);
+
+        props.sessionHandler();
```

Comment:
```suggestion
        resetPrint();

        const request = registration.getSubmitRequest(state.flowID, state.csrf);
        updateState({ request: request });

        const jsonRes = await API.makeRequest(registration.stringifyBody(request));
        updateState({
            accountKey: jsonRes.json.identity.traits.accountKey,
            sessionID: jsonRes.json.session.id,
            request: request,
            response: { headers: jsonRes.headers, status: jsonRes.status, body: jsonRes.json },
        });

        props.sessionHandler();
```

The name `sessionHandler` could be more expressive. Something like `registrationSuccessCallback` or something like that.

Reply:
I propose "checkForActiveSessionCallback" :D